crates/weaver_forge/src/config.rs

```diff
@@ -387,40 +309,446 @@ impl CaseConvention {
     }
 }
 
-impl TargetConfig {
-    pub(crate) fn try_new(loader: &dyn FileLoader) -> Result<TargetConfig, Error> {
-        let weaver_file = loader
-            .load_file(WEAVER_YAML)
-            .map_err(|e| InvalidConfigFile {
-                config_file: WEAVER_YAML.into(),
-                error: e.to_string(),
-            })?;
-        if let Some(weaver_file) = weaver_file {
-            serde_yaml::from_str(&weaver_file).map_err(|e| InvalidConfigFile {
-                config_file: WEAVER_YAML.into(),
-                error: e.to_string(),
-            })
-        } else {
-            Ok(TargetConfig::default())
+impl Default for WeaverConfig {
+    fn default() -> Self {
+        Self {
+            file_name: Default::default(),
```

Comment:
Nit: Seeing this again, in the future, we can take file, function, arg, struct, field naming conventions config and put them in a (flattened) nested struct.

Reply:
It's an option indeed. More generally, I was wondering if this approach would be used by the authors of code generation, or if these authors would directly use conversion filters that are semantically agnostic to the target language. That is, using filters like `snake_case` or `screaming_name_case` directly, instead of a more semantic version of these filters such as `function_name` or `constant_name` (where the mapping is defined by these entries in the config). I initially thought it was a good idea; I'm just not sure if it will be used. What do you think?

Reply:
Personal preference (and what I've seen so far) is to directly use the filters like `snake_case`.

I think we could back out this feature set and add it back in if we get requests for it.  Better to maintain less things with higher composability.